model/SearchEngine/Driver/Elasticsearch/QueryBuilder.php

```diff
@@ -207,24 +214,66 @@ private function buildConditionsByType(string $type, array $blocks): array
         return $this->getResourceConditions($blocks);
     }
 
+    private function containsLogicalModifier(string $block) {
+        $found = false;
+        foreach(self::LOGIC_MODIFIERS as $key => $logicModifier) {
+            if(strpos($block, $logicModifier) !== false) {
+                $found = true;
+                break;
+            }    
+        }    
+        return $found;
+    }
+
+    private function buildLogicCondition(string $block) {
+        $condition = '';
+        if(strpos($block, self::LOGIC_MODIFIERS['and'])) {
+            $logicBlocks = preg_split('/( '.self::LOGIC_MODIFIERS['and'].' )/i', $block);
+            $conditions = array_map([$this, 'buildConditionFromTheBlock'], $logicBlocks);
+            $condition = sprintf('(%s)', implode(' AND ', $conditions));
+        } elseif (strpos($block, self::LOGIC_MODIFIERS['or'])) {
+            $logicBlocks = preg_split('/( '.self::LOGIC_MODIFIERS['or'].' )/i', $block);
+            $conditions = array_map([$this,'buildConditionFromTheBlock'], $logicBlocks);
+            $condition = sprintf('(%s)', implode(' OR ', $conditions));
+        } elseif (strpos($block, self::LOGIC_MODIFIERS['not'])) {
+            $logicBlocks = preg_split('/( ' . self::LOGIC_MODIFIERS['not'] . ' )/i', $block);
+            $conditions = array_map([$this,'buildConditionFromTheBlock'], $logicBlocks);
+            $condition = sprintf('NOT (%s)', implode(' OR ', $conditions));
+        }
+        return $condition;
+    }
```

Comment:
chore suggestions. Same as the other comment + other considerations:

- Please, notice that `strpos($block, self::LOGIC_MODIFIERS['and'])` will return 0, which will translate to `false` in the `if`. It is better to use strict comparison, like `=== false`
- Elseif is not really used as generally when you have it, means you could end the flow earlier. It also makes the code more cluttered. I am leaving suggestion for clear code.
 
```suggestion
    private function buildLogicCondition(string $block): ?string {
        if (strpos($block, self::LOGIC_MODIFIERS['and']) !== false) {
            $logicBlocks = preg_split('/( '.self::LOGIC_MODIFIERS['and'].' )/i', $block);
            $conditions = array_map([$this, 'buildConditionFromTheBlock'], $logicBlocks);
            
            return sprintf('(%s)', implode(' AND ', $conditions));
        }
        
        if (strpos($block, self::LOGIC_MODIFIERS['or']) !== false) {
            $logicBlocks = preg_split('/( '.self::LOGIC_MODIFIERS['or'].' )/i', $block);
            $conditions = array_map([$this,'buildConditionFromTheBlock'], $logicBlocks);

            return sprintf('(%s)', implode(' OR ', $conditions));
        } 
        
        if (strpos($block, self::LOGIC_MODIFIERS['not']) !== false) {
            $logicBlocks = preg_split('/( ' . self::LOGIC_MODIFIERS['not'] . ' )/i', $block);
            $conditions = array_map([$this,'buildConditionFromTheBlock'], $logicBlocks);

            return sprintf('NOT (%s)', implode(' OR ', $conditions));
        }
        
        return null;
    }
```